### PR TITLE
prov/gni: improve some warn statements

### PR DESCRIPTION
**Pull request by hppritcha**
Use strerror in FI_EP_MSG connection setup in GNI provider
to help with debugging.  This is more useful than just printing
error return code.

@sungeunchoi 

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

File: prov/gni/src/gnix_cm.c [hunk 1]
```diff
@@ -281,8 +281,8 @@ int _gnix_ep_progress(struct gnix_fid_ep *ep)
 			}
 		} else {
 			GNIX_FATAL(FI_LOG_EP_CTRL,
-				   "Unexpected read size: %d\n",
-				   bytes_read);
+				   "Unexpected read size: %d err: %s\n",
+				   bytes_read,strerror(errno));
```
Comment (chuckfossen) [inline prov/gni/src/gnix_cm.c]: add a space after the ,

**Review by chuckfossen — APPROVED** [review]
looks good

**Comment by sungeunchoi** [comment]
Please see #904.  I don't know if errno will be preserved correctly here.

File: prov/gni/src/gnix_cm.c (hunk 1 above)
Comment (hppritcha) [reply prov/gni/src/gnix_cm.c]: oh, I disabled my style checker.

**Comment by hppritcha** [comment]
@sungeunchoi good point, I added an ```errno_keep``` stack variable to trap the ```errno``` before diving in to ```GNIX_WARN```

**Review by sungeunchoi — APPROVED** [review]

**Pull request merged**